eng/pipelines/global-build.yml

```diff
@@ -130,8 +130,9 @@ jobs:
     jobTemplate: /eng/pipelines/common/global-build-job.yml
     buildConfig: release
     platforms:
-    - Linux_x64
+    - SourceBuild_Linux_x64
     jobParameters:
       nameSuffix: SourceBuild
-      buildArgs: /p:DotNetBuildFromSource=true
+      buildArgs: -subset clr+libs+host+packs /p:DotNetBuildFromSource=true
+      buildingOnSourceBuildImage: true
```

Comment:
Given that the platform already has `SourceBuild` prefix, it might make sense to move this parameter `buildingOnSourceBuildImage` to the platform-matrix expansion?